src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -3231,6 +3231,9 @@ def update_functionapp_app_service_plan(cmd, instance, sku=None, number_of_worke
     if number_of_workers is not None:
         number_of_workers = validate_range_of_int_flag('--number-of-workers / --min-instances',
                                                        number_of_workers, min_val=0, max_val=20)
+    if is_plan_flex(cmd, instance):
+        return update_flex_app_service_plan(instance)
```

Comment:
The `instance` here is the requested ServerFarm information from the user right?

Reply:
Yes, it is done in a separate method. Command ensures that this method get the fully populated object (logic is in command definition). The job of this method is just to modify the object.